### DISCUSSION TITLE
OpenAI announces structured outputs

### DISCUSSION THREAD

**Discussion by legaltextai** [freelawproject/courtlistener#4291] [General]
They now [guarantee](https://openai.com/index/introducing-structured-outputs-in-the-api/?utm_source=thetechoasis.beehiiv.com&utm_medium=newsletter&utm_campaign=a-self-correcting-humanoid-grokking-a-tool-that-works&_bhlid=cf03f5e38a3be218bd8cddafcb4986507c2e774b) the model’s response will conform to a particular schema. 

They did have an option for json output before but without the guarantee part. 

Why is it a big deal? You can now instruct the model to analyze any case or thousands of cases and return the output in a structured json as per your requirements. For example, you can add these template schemas to the model instruction. 
{
  "Issue": "",
  "Rule": "",
  "Application": "",
  "Conclusion": ""
}

{
  "cited_case": "",
  "treatment": "positive/negative/neutral"
}

And then set up search query to apply these as filters. "Find me the case that considered the issue of such and such...", "Find me cases after 2022 that reviewed 141 S. Ct. 1183 and summarize its treatment"

**Comment by mlissner** [comment]
I saw this. I thought it looked really useful for our scanning project too, assuming this works with their vision API.
